portal/integrated_windows_authentication/src/main/java/com/esri/samples/integrated_windows_authentication/IntegratedWindowsAuthenticationController.java

```diff
@@ -0,0 +1,202 @@
+      iwaSecuredPortal = new Portal(portalUrlTextField.getText(), true);
+
+      // clear any existing items in the list view
+      resultsListView.getItems().clear();
+
+      // clear the information about the previously loaded map
+      loadWebMapTextView.setText("");
+
+      // show portal load state
+      progressIndicator.setVisible(true);
+      loadStateTextView.setText("Searching for web map items on the portal at " + iwaSecuredPortal.getUri());
+
+      // load the portal items
+      iwaSecuredPortal.loadAsync();
+      iwaSecuredPortal.addDoneLoadingListener(() -> {
+        if (iwaSecuredPortal.getLoadStatus() == LoadStatus.LOADED) {
+          try {
+            // update load state in UI with the portal URI
+            loadStateTextView.setText("Connected to the portal on " + new URI(iwaSecuredPortal.getUri()).getHost());
+          } catch (URISyntaxException e) {
+            new Alert(Alert.AlertType.ERROR, "Error getting URI from portal: " + e.getMessage()).show();
+          }
+
+          // report the user name used for this connection
+          if (iwaSecuredPortal.getUser() != null) {
+            loadStateTextView.setText("Connected as: " + iwaSecuredPortal.getUser().getUsername());
+          } else {
+            // if connecting to an unsecured portal, no user credentials are needed to authenticate access
+            loadStateTextView.setText("Connected as: Anonymous");
+          }
+
+          // search the portal for web maps
+          ListenableFuture<PortalQueryResultSet<PortalItem>> portalItemResultFuture = iwaSecuredPortal.findItemsAsync(new PortalQueryParameters("type:(\"web map\" NOT \"web mapping application\")"));
+          portalItemResultFuture.addDoneListener(() -> {
+            try {
+              // get the result
+              PortalQueryResultSet<PortalItem> portalItemSet = portalItemResultFuture.get();
+              List<PortalItem> portalItems = portalItemSet.getResults();
+              // add the IDs and titles of the portal items to the list view
+              portalItems.forEach(portalItem -> resultsListView.getItems().add(new Pair<>(portalItem.getItemId(), portalItem.getTitle())));
+
+            } catch (ExecutionException | InterruptedException e) {
+              new Alert(Alert.AlertType.ERROR, "Error getting portal item set from portal: " + e.getMessage()).show();
+            }
+            // hide the progress indicator
+            progressIndicator.setVisible(false);
+          });
+
+        } else {
+          // hide the progress indicator and reset the load state text
+          progressIndicator.setVisible(false);
+          loadStateTextView.setText("");
+          // report error
+          new Alert(Alert.AlertType.ERROR, "Portal sign in failed: " + iwaSecuredPortal.getLoadError().getCause().getMessage()).show();
+        }
+      });
+    } else {
+      new Alert(Alert.AlertType.ERROR, "Portal URL is empty. Please enter a portal URL.").show();
```

Comment:
Move this to the first case of the `if` statement since it's shorter. Way down here, you lose track of what the `if` condition was.